subsys/disk/disk_access.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <init.h>
+#include <disk_access.h>
+#include <errno.h>
+#include <device.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_DISK_LEVEL
+#include <logging/sys_log.h>
+
+/* list of mounted file systems */
+static sys_dlist_t disk_access_list;
+
+/* lock to protect storage layer registration */
+static struct k_mutex mutex;
+
+struct disk_info *disk_access_get_di(const char *name)
+{
+	struct disk_info *disk = NULL, *itr;
+	size_t name_len = strlen(name);
+	sys_dnode_t *node;
+
+	k_mutex_lock(&mutex, K_FOREVER);
+	SYS_DLIST_FOR_EACH_NODE(&disk_access_list, node) {
+		itr = CONTAINER_OF(node, struct disk_info, node);
+
+		/*
+		 * Move to next node if mount point length is
+		 * shorter than longest_match match or if path
+		 * name is shorter than the mount point name.
+		 */
+		if (strlen(itr->name) != name_len) {
+			continue;
+		}
+
+		/* Check for disk name match */
+		if (strncmp(name, itr->name, name_len) == 0) {
```

Comment:
Will this be an issue if we have two disks with the same prefix?

Reply:
we can not have two disk with same name

Reply:
?